test/threads_exec.jl

```diff
@@ -671,6 +671,68 @@ let ch = Channel{Char}(0), t
     @test String(collect(ch)) == "hello"
 end
 
+@testset "@spawn interpolation" begin
+    # Issue #30896: evaluating argumentss immediately
+    begin
+        outs = zeros(5)
+        @sync begin
+            local i = 1
+            while i <= 5
+                Threads.@spawn setindex!(outs, $i, $i)
+                i += 1
+            end
+        end
+        @test outs == 1:5
+    end
+
+    # Args
+    @test fetch(Threads.@spawn 2+$2) == 4
+    @test fetch(Threads.@spawn Int($(2.0))) == 2
+    a = 2
+    @test fetch(Threads.@spawn *($a,$a)) == a^2
+    # kwargs
+    @test fetch(Threads.@spawn sort($([3 2; 1 0]), dims=2)) == [2 3; 0 1]
+    @test fetch(Threads.@spawn sort([3 $2; 1 $0]; dims=$2)) == [2 3; 0 1]
+
+    # Supports multiple levels of interpolation
+    @test fetch(Threads.@spawn :($a)) == a
```

Comment:
This confused me briefly maybe replace with:

```
@test fetch(Threads.@spawn :($a + a)) == Expr(:call, :+, a, :a)
```

Reply:
I've replaced it with what I think is overall a clearer test that actually showcases the modification property as well.

Does this look okay now?:
https://github.com/JuliaLang/julia/pull/33119/files#diff-d94a81ec91d1b6e2ba2cdb900b30cb3dR699